### PR TITLE
Add `--skip-keys` option

### PR DESCRIPTION
**Pull request by andre-rosa**

### CODE REVIEW

File: superflore/generators/bitbake/gen_packages.py [hunk 1]
```diff
@@ -35,8 +35,12 @@
 
 
 def regenerate_installer(
-    overlay, pkg, distro, preserve_existing, tar_dir, md5_cache, sha256_cache
+    overlay, pkg, distro, preserve_existing, tar_dir, md5_cache, sha256_cache,
+    skip_keys
 ):
+    if pkg in skip_keys:
+        warn("Package '%s' is on skip-keys, skipping..." % pkg)
```
Comment (allenh1) [inline superflore/generators/bitbake/gen_packages.py]: I'm not a huge fan of the phrasing here. Maybe something like `"Package '%s' is in skip-keys list, skipping..."`? I'm not sure.

File: superflore/generators/ebuild/gen_packages.py [hunk 2]
```diff
@@ -42,7 +42,10 @@
 org_license = "BSD"
 
 
-def regenerate_pkg(overlay, pkg, distro, preserve_existing=False):
+def regenerate_pkg(overlay, pkg, distro, preserve_existing, skip_keys):
```
Comment (allenh1) [inline superflore/generators/ebuild/gen_packages.py]: I don't remember what the implication of `preserve_existing` not defaulting to `False` is, but would probably feel better if it continued to default to `False`.

File: superflore/parser.py [hunk 3]
```diff
@@ -59,4 +59,10 @@ def get_parser(tool_tip, is_generator=True):
             help='location of the upstream repository',
             type=str
         )
+        parser.add_argument(
+            '--skip-keys',
+            nargs='+',
+            help='The specified rosdep keys will be ignored. '
```
Comment (allenh1) [inline superflore/parser.py]: I don't think "rosdep keys" is the correct term. I think this should read `'packages to skip during regeneration'`. If you intend to skip any packages depending on a certain `rosdep` key, I think that is already handled (at least it is on Gentoo).

File: superflore/generators/bitbake/gen_packages.py (hunk 1 above)
Comment (andre-rosa) [reply superflore/generators/bitbake/gen_packages.py]: Done, thanks.

File: superflore/generators/ebuild/gen_packages.py (hunk 2 above)
Comment (andre-rosa) [reply superflore/generators/ebuild/gen_packages.py]: No worries; done.

File: superflore/parser.py (hunk 3 above)
Comment (andre-rosa) [reply superflore/parser.py]: Done

File: superflore/generators/bitbake/gen_packages.py [hunk 4]
```diff
@@ -35,8 +35,12 @@
 
 
 def regenerate_installer(
-    overlay, pkg, distro, preserve_existing, tar_dir, md5_cache, sha256_cache
+    overlay, pkg, distro, preserve_existing, tar_dir, md5_cache, sha256_cache,
+    skip_keys=None
 ):
+    if skip_keys and pkg in skip_keys:
```
Comment (allenh1) [inline superflore/generators/bitbake/gen_packages.py]: Sorry, I didn't realize this until just now... Can we handle this `if` statement in the caller of this function? I think that would make it harder to forget on any platforms that get added in the future.

Comment (andre-rosa) [reply superflore/generators/bitbake/gen_packages.py]: Sure; could you please take a look?

**Comment by allenh1** [comment]
@tfoote I'm going to run a quick test locally, then I'll merge this.

**Comment by andre-rosa** [comment]
Thank you for reviewing! I'll rebase the other changes on top.

**Comment by andre-rosa** [comment]
Closes: #170

**Pull request merged**